Services/Database/classes/PDO/Manager/class.ilDBPdoManager.php

```diff
@@ -471,4 +471,75 @@ public function dropTable(string $name): bool
 
         return (bool) $this->pdo->exec("DROP TABLE $name");
     }
+
+    /**
+     * @throws ilException
+     */
+    public function addForeignKey(
```

Comment:
We must accept a collection of fields/columns here which can be referenced, because a primary key in the referenced table can be a combined key.